DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/RawEventData.h

```diff
@@ -0,0 +1,95 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+//
+//file RawEventData.h class  for RAW data format
+// Artur.Furs
+//
+#ifndef ALICEO2_FIT_RAWEVENTDATA_H_
+#define ALICEO2_FIT_RAWEVENTDATA_H_
+
+#include "DataFormatsFT0/Digit.h"
+#include "Headers/RAWDataHeader.h"
+#include "DataFormatsFT0/LookUpTable.h"
+#include <CommonDataFormat/InteractionRecord.h>
+#include <iostream>
+#include <bitset>
+#include "Rtypes.h"
+namespace o2
+{
+namespace ft0
+{
+constexpr int NCHANNELS_FT0 = 208;
+constexpr int NCHANNELS_PM = 12;
+constexpr int NPMs = 18;
+constexpr int NBITS_EVENTDATA = 9;
+
+struct EventHeader {
+  uint startDescriptor : 4;
+  uint Nchannels : 4;
+  uint reservedField : 32;
+  uint orbit : 32;
+  uint bc : 12;
+  ClassDefNV(EventHeader, 1);
+};
```

Comment:
Also note that your header is 128 bits long (from which you use 84). So, you cannot put it in a single GBT word. You should either reduce it to 64 bits by shortening the reservedField and rearranging to 
````
struct EventHeader {
  uint startDescriptor : 4;
  uint Nchannels : 4;
  uint reservedField : 12;
  uint bc : 12;
  uint orbit : 32;
  ClassDefNV(EventHeader, 1);
};
````
or split to 2 GBT words.

Reply:
when I print sizeof (EventHeader) and (2*EventData[0])  I see
eventHeader 16 data 16
what wrong?

Reply:
But you were saying that it is 80 bits long. The GBT word is 80 bits, if T0 wants to transmit the 128 bit long header to the CRU, it should do it in 2 80 bits words, then on the CRU each of these words will be padded by 0s to 128 bits. If your aim to create raw data as they come out from the CRU, you should reproduce this.

Reply:
You are right. I set reservedField=28, now sum of bits = 80

Reply:
still not enough: the 1st 3 fields add up to 36 bits so they cannot be embedded to 1 32 bit words, therefore the reservedField will start new uint. The only way to make the useful payload to occupy 1st 80 bits is to reshuffle the header like I showed above. Or to split it to 2 GBT words.

Reply:
Sorry, it was typo. Detector field is 28bits,  in sum 80bit in header

Reply:
I will think...

Reply:
Actually, also this layout will allow to have 28 bits reserved and to fit the payload to 80 bits:
````
struct EventHeader {
  uint startDescriptor : 4;
  uint reservedField : 28;
  uint orbit : 32;
  uint bc : 12;
  uint Nchannels : 4;
  ClassDefNV(EventHeader, 1);
};
````

Reply:
yes, I just asked Dima about this changes. Thank you